src/sentry/incidents/grouptype.py

```diff
@@ -27,29 +38,142 @@ class MetricIssueEvidenceData(EvidenceData):
+                condition_group=self.detector.workflow_condition_group, condition_result=priority
+            )
+        except DataCondition.DoesNotExist:
+            logger.exception(
+                "Failed to find detector trigger, cannot create metric issue occurrence",
+                extra={"detector_id": self.detector.id},
+            )
+            return None
+
+        try:
+            query_subscription = QuerySubscription.objects.get(id=data_packet.source_id)
+        except QuerySubscription.DoesNotExist:
+            logger.exception(
+                "Failed to find query subscription related to the detector, cannot create metric issue occurrence",
+                extra={"detector_id": self.detector.id},
+            )
+            return None
+
+        try:
+            snuba_query = SnubaQuery.objects.get(id=query_subscription.snuba_query_id)
+        except SnubaQuery.DoesNotExist:
+            logger.exception(
+                "Failed to find snuba query related to the detector, cannot create metric issue occurrence",
+                extra={"detector_id": self.detector.id},
+            )
+            return None
+
+        try:
+            assignee = parse_and_validate_actor(
+                self.detector.created_by_id, self.detector.project.organization_id
+            )
+        except Exception:
+            assignee = None
+
+        title = self.construct_title(snuba_query, detector_trigger, priority)
+        event_data = {
+            "project_id": self.detector.project_id,
+            "environment": self.detector.config.get("environment"),
+            "platform": None,
+            "sdk": None,
+        }  # XXX: may need to add to this
+
+        return (
+            DetectorOccurrence(
+                issue_title=self.detector.name,
+                subtitle=title,
+                resource_id=None,
+                evidence_data={
+                    "detector_id": self.detector.id,
+                    "value": priority,
+                    "data_condition_ids": [detector_trigger.id],
+                    "data_condition_type": detector_trigger.type,
+                    "data_condition_comparison_value": self.extract_value(data_packet),
+                    "alert_id": alert_id,
+                },
+                evidence_display=[],  # XXX: may need to pass more info here for the front end
+                type=MetricIssue,
+                level="error",
+                culprit="",
+                priority=priority,
+                assignee=assignee,
+            ),
+            event_data,
         )
-        return occurrence, {}
 
     def extract_dedupe_value(self, data_packet: DataPacket[QuerySubscriptionUpdate]) -> int:
         return int(data_packet.packet.get("timestamp", datetime.now(UTC)).timestamp())
 
     def extract_value(self, data_packet: DataPacket[QuerySubscriptionUpdate]) -> int:
         return data_packet.packet["values"]["value"]
 
+    def construct_title(
```

Comment:
This is based off of https://github.com/getsentry/sentry/blob/master/src/sentry/incidents/utils/metric_issue_poc.py#L52 - I thought about refactoring it but it seemed not great to have it accept either an `AlertRule` or a `DataCondition` so I just reused it knowing we'll delete the other one.

Reply:
🎉 nice - i think overall this is a great example of how we can utilize the detector class to add helper methods.